src/main/groovy/edu/oregonstate/mist/idimages/resources/IdImagesResource.groovy

```diff
@@ -35,21 +37,46 @@ class IdImagesResource extends Resource {
     @Produces("image/jpg")
     @Timed
     public Response getByOSUID(@Auth AuthenticatedUser _, @PathParam('id') String id) {
+        Integer bannerPIDM = idImageDAO.getPIDM(id)
+
+        if (!bannerPIDM) {
+            return notFound().type(MediaType.APPLICATION_JSON).build()
+        }
         Blob image = idImageDAO.getByID(id)
 
+        //Return a placeholder image if a person exists but doesn't have an ID image
         if (!image) {
-            return notFound().type(MediaType.APPLICATION_JSON).build()
+            return ok(getImageStream(ImageIO.read(new File("images/defaultImage.jpg")))).build()
         }
+        //Return an ID image of a person
         try {
-            BufferedImage buffImg = ImageIO.read(image.getBinaryStream())
-            ByteArrayOutputStream outputStream = new ByteArrayOutputStream()
-            ImageIO.write(buffImg, "jpg", outputStream)
-            byte[] imageData = outputStream.toByteArray()
-            Response.ok(imageData).build()
+            Response.ok(getImageStream(ImageIO.read(image.getBinaryStream()))).build()
```

Comment:
This line has about 5 method calls. It will work as is, but when  you start getting a lot of method calls with method calls within them it is usually a sign that it could be refactored a little bit.

Why not just pass the image variable to getImageStream ? Also since this is groovy, instead of using: image.getBinaryStream() you can use: image.binaryStream , but that alone is not enough.